subsys: disk: ram: Make RAM disk size be configurable

pfalcon opened the pull request:
Hardcoded 96KB starts to overload RAM regions and fail CI tests.
Quick test shows that 80KB ramdisk is ok, (passes tests/posix/fs).
And of course, targets with wealth of RAM may want to use bigger
ramdisks.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon committed:
sub-sys: disk: ram: Make RAM disk size be configurable

Hardcoded 96KB starts to overload RAM regions and fail CI tests.
Quick test shows that 80KB ramdisk is ok, (passes tests/posix/fs).
And of course, targets with wealth of RAM may want to use bigger
ramdisks.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon commented:
Dunno how we survived for so long without configurable ramdisk. As mentioned, starts to fail CI for me, e.g. https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/33594/1/tests

Apparently, everyone was scared away by that comment that 96K is the min. I for sure look not for the 1st time into it. Turns out to be an urban legend.

andrewboie commented on subsys/disk/disk_access_ram.c:
```diff
@@ -20,10 +20,10 @@
  */
 #include "fat12_ramdisk.h"
 #else
-/* A 96KB RAM Disk, which meets ELM FAT fs's minimum block requirement. Fit for
```
I don't think this comment should be deleted.
Is there a Kconfig which indicates the filesystem in use? Seems like if the size is made smaller than 96 the fs will be invalid

pfalcon replied:
As written in the commit message, I tested tests/posix/fs with 80K and it passed. So, we either have an fs test which doesn't detect invalid fs, we have another fs rather than fatfs by default, or that comment isn't exactly right. Can make bets which is it ;-)

pfalcon replied:
> As written in the commit message, I tested tests/posix/fs with 80K and it passed.

To prove the point, I cherry-picked that change from https://github.com/zephyrproject-rtos/zephyr/pull/12984, as it's going to hit on master soon anyway.

pfalcon committed:
tests: posix: fs: Decrease ramdisk size, increase stack size

This test seems to be fine in the current master, but doing
development, it easily starts to overflow RAM of some boards and/or
crash due to stack checks. So, decrease the ramdisk (the biggest
eater of RAM here) from default 96K to 80K, and bump main stack
size proactively.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>
